lightning/src/util/ser.rs

```diff
@@ -694,18 +723,23 @@ macro_rules! impl_for_vec {
 impl Writeable for Vec<u8> {
 	#[inline]
 	fn write<W: Writer>(&self, w: &mut W) -> Result<(), io::Error> {
-		(self.len() as u16).write(w)?;
+		U16Or64(self.len() as u64).write(w)?;
```

Comment:
Should be concerned with writing an incompatible length for an LN message? Or are we leaving that to the caller to enforce / assume will never happen?